lib/SILOptimizer/Analysis/MemoryBehavior.cpp

```diff
@@ -242,13 +242,9 @@ MemBehavior MemoryBehaviorVisitor::visitLoadInst(LoadInst *LI) {
   if (!mayAlias(LI->getOperand()))
     return MemBehavior::None;
```

Comment:
@eeckstein one thing we need to think about is do we consider from a membehavior perspective that the retain from a load [copy] is an issue as well in terms of side-effects. I imagine probably not.

Reply:
Right, it is not. Therefore the original version of visitLoad() is correct.

Reply:
Exactly the sort of thing that needs to be specified in the definition of MemBehavior: writes to non-programmer visible memory, particularly refcounts, are not considered write or side effects.

Reply:
If we are going to do that, we should model retain behavior in MemBehavior explicitly. That would be the right way to do that.

Reply:
I don't think that this is true. If you look in SILNodes.def we consider retain to have side-effects since we don't model retains separately in MemBehavior. See:

https://github.com/apple/swift/blob/f58c14335518e611aa8b8a29aafe9ea73848f919/include/swift/SIL/SILNodes.def#L793

I agree with you if we model retains separately in membehavior.